### DISCUSSION TITLE
Debug hexdump stops working nicely

### DISCUSSION THREAD

**Discussion by StevenCellist** [jgromes/RadioLib#862] [General]
Kind of an issue but actually not at all...
After a whole bunch of uplinks with debug mode enabled, my hexdump all of a sudden isn't showing utf-8 translation anymore. At one point, it switched to @ and stayed at that. Here's the break point:
```
17:46:06.261 > Channel frequency UL = 867.299927 MHz
17:46:06.269 > DR 26: LORA (SF: 7, BW: 125.000000, CR: 7)
17:46:06.272 > uplinkMsg pre-MIC:
17:46:06.273 > 0000000 01 ea 39 01 53 10 05 d2 ad 84 65 01 aa ea 0b 26 | ..9.S.....e....&
17:46:06.280 > 0000010 40 aa ea 0b 26 80 25 01 0a cf 90 6e 28 11 52 16 | @...&.(.R.
17:46:06.287 > 0000020 55 e8 f3 6c a3 20 d1 05 1d 53 4c 0b 39 95       | @@@@@@@@@@@@@@
17:46:06.293 > uplinkMsg:
17:46:06.294 > 0000000 49 00 00 00 00 00 aa ea 0b 26 25 01 00 00 00 1a | @@@@@@@@@@@@@@@@
17:46:06.300 > 0000010 40 aa ea 0b 26 80 25 01 0a cf 90 6e 28 11 52 16 | @@@@@@@@@@@@@@@@
17:46:06.307 > 0000020 55 e8 f3 6c a3 20 d1 05 1d 53 0e 3a d4 a0       | @@@@@@@@@@@@@@    
17:46:06.313 > Timeout in 138624 us
17:46:06.413 > Uplink sent <-- Rx Delay start
```

No clue how to reproduce (probably significant uptime and output), but thought I'd mention it for the sake of it.. who knows

**Comment by jgromes** [comment]
The hexdump method is based on ESP8266 implementation of printf(): https://github.com/esp8266/Arduino/blob/65579d29081cb8501e4d7f786747bf12e7b37da2/cores/esp8266/Print.cpp#L50, so maybe this is a known issue in the ESP8266 Arduino world.

I haven't encountered this problem yet, though I usually use hexdump for short-term debugging, so if there is something dependent on long runtime (e.g. small memory leak), I wouldn't have noticed it.

**Reply by StevenCellist** [reply]
Yeah I don't care in the slightest actually, just so you know it's there. You can close this if you have no interest in finding the culprit
